tls/s2n_early_data.c

```diff
@@ -96,7 +96,7 @@ static S2N_RESULT s2n_early_data_validate(struct s2n_connection *conn)
     const size_t app_protocol_size = strlen(conn->application_protocol);
     if (app_protocol_size > 0 || config->application_protocol.size > 0) {
         RESULT_ENSURE_EQ(config->application_protocol.size, app_protocol_size + 1 /* null-terminating char */);
-        RESULT_ENSURE_EQ(memcmp(config->application_protocol.data, conn->application_protocol, app_protocol_size), 0);
+        RESULT_ENSURE_EQ(s2n_constant_time_equals(config->application_protocol.data, (uint8_t *) conn->application_protocol, app_protocol_size), true);
```

Comment:
nit
```suggestion
        RESULT_ENSURE(s2n_constant_time_equals(config->application_protocol.data, (uint8_t *) conn->application_protocol, app_protocol_size));
```

Reply:
```RESULT_ENSURE``` requires an error if ```cond``` is not satisfied. Use ```S2N_ERR_SAFETY``` as the error.